### PR TITLE
chore: add linter for OpenAPI

### PR DESCRIPTION
**Pull request by thomaspoignant**
Add linter for OpenAPI validation.

### CODE REVIEW

File: .,
Comment (Kavindu-Dodan) [inline .,:1]: I think this is not needed

Comment (thomaspoignant) [reply .,:1]: definitely not, I am not sure why it is here 🤦

File: .github/workflows/lint-openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      - synchronize
+
+jobs:
+  openAPILint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: OpenAPI Lint Checks
+        uses: nwestfall/openapi-action@v1.0.2
```
Comment (Kavindu-Dodan) [inline .github/workflows/lint-openapi.yaml:17]: Regarding this action, it seems this has been unmaintained for some time. And I checked alternatives and came across redocly-cli (MIT license) and used their docker image [1] to perform a similar validation 

You can see my attempt in the fork [2] and the test GH action failure run in my fork [3]

Let me know your thoughts on this :) 

[1] - https://github.com/Redocly/redocly-cli?tab=readme-ov-file#docker
[2] - https://github.com/open-feature/protocol/compare/test-validate
[3] - https://github.com/open-feature/protocol/actions/runs/8913421964/job/24478855581

**Comment by thomaspoignant** [comment]
@Kavindu-Dodan Maybe we should take your branch instead of this one.
When it comes to the validation I am not sure why it fails, but this is worth checking.

File: .github/workflows/lint-openapi.yaml (hunk 1 above)
Comment (beeme1mr) [reply .github/workflows/lint-openapi.yaml:17]: This is another option that adds lint violations as annotations in the PR.

https://github.com/stoplightio/spectral-action

**Comment by Kavindu-Dodan** [comment]
> @Kavindu-Dodan Maybe we should take your branch instead of this one. When it comes to the validation I am not sure why it fails, but this is worth checking.

I will take a shot on this then :)

**Comment by Kavindu-Dodan** [comment]
@thomaspoignant here is a PR with redocly - https://github.com/open-feature/protocol/pull/15

File: .github/workflows/lint-openapi.yaml (hunk 1 above)
Comment (Kavindu-Dodan) [reply .github/workflows/lint-openapi.yaml:17]: @beeme1mr checked but I am not super happy yet as there are open issues and PRs that were not addressed recently. We can switch back if we see issues with redocly

**Comment by thomaspoignant** [comment]
Close this PR in favor of #15

**Pull request closed without merging**